### PR TITLE
Initialised global gorm db variable

### PR DESCRIPTION
**Pull request by pkpraveen603**

### CODE REVIEW

File: cmd/main.go [hunk 1]
```diff
@@ -1,7 +1,29 @@
...
 func main()  {
+
+	//assign connection to global *gorm.DB variable DB
+	dbstring := config.DbURL(config.BuildConfig())
+	config.DB,err = gorm.Open("mysql", dbstring)
+	if err != nil {
+		log.Fatalf("Gorm: failed to open DB: %v\n", err)
+	}
+
+	defer func() {
```
Comment (mukesh0513) [inline cmd/main.go]: why are we closing this connection? we need to use for querying right?

Comment (mukesh0513) [reply cmd/main.go]: write a separate method for initialising db. code will be cleaner.

**Comment by mukesh0513** [comment]
Better pr names from next time.

File: cmd/main.go [hunk 2]
```diff
@@ -1,7 +1,31 @@
...
+	"Tradeasy/config"
+	"Tradeasy/internal/router"
+	"github.com/jinzhu/gorm"
+	"log"
+)
+
+var err error
+
+// initialiseDB ...assign connection to global *gorm.DB variable DB
+func initialiseDB() error{
```
Comment (mukesh0513) [inline cmd/main.go]: move this method to config file itself and run gofmt.

File: config/database.go [hunk 3]
```diff
@@ -2,11 +2,17 @@ package config
...
 	"fmt"
+	"github.com/jinzhu/gorm"
 	_ "github.com/kelseyhightower/envconfig"
 	"gopkg.in/yaml.v2"
 	"os"
 )
 
+var (
+	DB  *gorm.DB
+	err error
```
Comment (mukesh0513) [inline config/database.go]: why error is being initialised as global variable?

Comment (pkpraveen603) [reply config/database.go:13]: See line 66. If i keep it local then can't assign to DB variable which is global and err which would be local

**Review by mukesh0513 — APPROVED** [review]
LGTM

**Pull request merged**